### PR TITLE
Clarify `model_serializer(mode="wrap")` usage in documentation

### LINKED ISSUE #9987: [Documentation] `docs/errors/usage_errors.md` incorrectly reference `model_serializer` in the `field-serializer-signature` section
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Hello Dear Pydantic maintainers. I hope you are well in this hot summer (if you're up north)

Something is bothering me with the documentation, I must say.

It seems to me that this section of the documentation https://docs.pydantic.dev/2.8/errors/usage_errors/#field-serializer-signature heavily mentions `model_serializer` while it should actually mention `field_serializer`.

If i interpret this section correctly, everything seems right *except* the term "model_serializer". 

Would a basic string replacement s/model_serializer/field_serializer/ be enough ?

Should I submit a patch in this way ?

Also, there's no such (valuable) information in the `model-serializer-signature` section.
Would you be interested in an another update to fill this blank ?

I have found a lot of documentation for the `model_serializer(mode='plain')`, but either the documentation does not give examples for `mode='wrap'`, either I did not find them.

If there is indeed some missing bits, would be interested in me contributing to this as well? 
It really bothered me a few weeks ago, because I had to look inside the source code to find the matching types for the `handler` and `info` parameters to satisfy mypy.

Thank you for your work, pydantic is really helpful to me on a day to day basis.

### Example Code

```Python
# Does not apply.
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.8.2
        pydantic-core version: 2.20.1
          pydantic-core build: profile=release pgo=true
                 install path: /home/anvil/.local/lib/python3.12/site-packages/pydantic
               python version: 3.12.4 (main, Jun  7 2024, 00:00:00) [GCC 14.1.1 20240607 (Red Hat 14.1.1-5)]
                     platform: Linux-6.8.9-300.fc40.x86_64-x86_64-with-glibc2.39
             related packages: pyright-1.1.348 typing_extensions-4.9.0 email-validator-2.1.0.post1 mypy-1.10.0
                       commit: unknown

I was looking at the main branch, for what it's worth.
```

### PR DESCRIPTION
**Pull request by Anvil**
## Change Summary

I'm trying to fix a few issues with regard to the documentation of the `model_serializer(mode="wrap")` case. I *think* there was a few typo. I've also wrapped a line in a python example in order to avoid an horizontal scrollbar.

I've tried to respect the style, but please feel free to comment. Also, I'm not native-english, so a grammar error is still possible.

I think the branch lacks a meaningful example to add to model_serializer docstring, but we can still discuss that, either here or in another PR.

## Related issue number

fix #9987 

## Checklist

* [X] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @sydney-runkle

### CODE REVIEW

**Comment by Anvil** [comment]
Pipeline have passed (no surprise I guess).

Please review

**Comment by sydney-runkle** [comment]
Hey @Anvil, thanks for the PR! I'll review first thing tomorrow 🚀 !

File: pydantic/functional_serializers.py [hunk 1]
```diff
@@ -338,6 +339,16 @@ def serialize_model(self):
...
 
+    Two signatures are supported for `mode='plain'`, which is the default:
+
+    - `(self)`
+    - `(self, info: SerializationInfo)`
+
+    And two other signatures for `mode='wrap'`:
+
+    - `(self, nxt: SerializerFunctionWrapHandler)`
+    - `(self, nxt: SerializerFunctionWrapHandler, info: SerializationInfo)`
```
Comment (sydney-runkle) [inline pydantic/functional_serializers.py:350]:
```suggestion
    Two signatures are supported for `mode='plain'`, which is the default:

    - `(self)`
    - `(self, info: SerializationInfo)`

    And two other signatures for `mode='wrap'`:

    - `(self, nxt: SerializerFunctionWrapHandler)`
    - `(self, nxt: SerializerFunctionWrapHandler, info: SerializationInfo)`

        See [Custom serializers](../concepts/serialization.md#custom-serializers) for more information.
```

**Review by sydney-runkle — COMMENTED** [review]
Great work!!

**Comment by sydney-runkle** [comment]
This looks awesome - thanks for the fixes. Going to go ahead and merge this. Feel free to open another PR with a more meaningful example in the `model_serializer` docstring. We really appreciate your help in improving our documentation 🚀!

**Review by sydney-runkle — APPROVED** [review]

**Comment by Anvil** [comment]
Sorry for not replying earlier - I was away. My pleasure! See you soon.

**Pull request merged**